Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -0,0 +1,157 @@
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "FiniteDifferenceSolver.H"
+#include "Utils/WarpXConst.H"
+#include <AMReX_Gpu.H>
+
+using namespace amrex;
+
+/**
+ * \brief Update the E field at the boundary, using the Silver-Mueller condition
+ */
+void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Efield,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& Bfield,
+    amrex::Box domain_box,
+    amrex::Real const dt ) {
+
+#ifdef WARPX_DIM_RZ
+    amrex::Abort("The Silver-Mueller boundary conditions cannot be used in RZ geometry.");
+#else
+
+    // Ensure that we are using the Yee solver
+    if (m_fdtd_algo != MaxwellSolverAlgo::Yee) {
+        amrex::Abort("The Silver-Mueller boundary conditions can only be used with the Yee solver.");
+    }
+
+    // Ensure that we are using the cells the domain
+    domain_box.enclosedCells();
+
+    // Calculate relevant coefficients
+    amrex::Real const cdt_over_dx = PhysConst::c*dt*m_stencil_coefs_x[0];
+    amrex::Real coef1_x = (1. - cdt_over_dx)/(1. + cdt_over_dx);
+    amrex::Real coef2_x = 2*cdt_over_dx/(1. + cdt_over_dx) / PhysConst::c;
+#ifdef WARPX_DIM_3D
+    amrex::Real const cdt_over_dy = PhysConst::c*dt*m_stencil_coefs_y[0];
+    amrex::Real coef1_y = (1. - cdt_over_dy)/(1. + cdt_over_dx);
+    amrex::Real coef2_y = 2*cdt_over_dy/(1. + cdt_over_dy) / PhysConst::c;
+#endif
+    amrex::Real const cdt_over_dz = PhysConst::c*dt*m_stencil_coefs_z[0];
+    amrex::Real coef1_z = (1. - cdt_over_dz)/(1. + cdt_over_dx);
+    amrex::Real coef2_z = 2*cdt_over_dz/(1. + cdt_over_dz) / PhysConst::c;
+
+    // Loop through the grids, and over the tiles within each grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    // tiling is usually set by TilingIfNotGPU()
+    // but here, we set it to false because of potential race condition,
+    // since we grow the tiles by one guard cell after creating them.
+    for ( MFIter mfi(*Efield[0], false); mfi.isValid(); ++mfi ) {
+
+        // Extract field data for this grid/tile
+        Array4<Real> const& Ex = Efield[0]->array(mfi);
+        Array4<Real> const& Ey = Efield[1]->array(mfi);
+        Array4<Real> const& Ez = Efield[2]->array(mfi);
+        Array4<Real> const& Bx = Bfield[0]->array(mfi);
+        Array4<Real> const& By = Bfield[1]->array(mfi);
+        Array4<Real> const& Bz = Bfield[2]->array(mfi);
+
+        // Extract the tileboxes for which to loop
+        Box tbx  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box tby  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box tbz  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // We will modify the first (i.e. innermost) guard cell
+        // (if it is outside of the physical domain)
+        // Thus, the tileboxes here are grown by 1 guard cell
+        tbx.grow(1);
```

Comment:
@WeiqunZhang Instead of doing this (growing the tiles themselves), could I modify the `for ( MFIter mfi(*Efield[0], false); mfi.isValid(); ++mfi ) {` to create tiles perfectly divide the local box **extended by one guard cells**, without the tiles overlapping each other.